### PR TITLE
ci: add semantic release for master

### LINKED ISSUE #30: Semantic release in project
Please, add a sematic release to project. 

The semantic release will be deployed from the master branch and publish core schemas to gh-pages with saving previous versions.

Also please add:
- commit analyze 
- changelog 
- automatic changes versions of releases schemas
- release to github - include only core schemas of release version

Core schemas included: 
- `collection.json`
- `deps.json`
- `linter.json`
- `package.json`
- `schemaver.json`

example of gh-pages: 
```
.
├── collection.json
├── deps.json
├── linter.json
├── package.json
├── schemaver.json
└── 1.0.0 /
|     ├── collection.json
|     ├── deps.json
|     ├── linter.json
|     ├── package.json
|     └── schemaver.json
└── ....
```
Where `1.0.0` is folder name of the version of schemas.

Recommendation:
- [travis](https://travis-ci.org)
- [semantic release](https://github.com/semantic-release/semantic-release) 
- [conventional commits](https://conventionalcommits.org)

### PR DESCRIPTION
**Pull request by romanrostislavovich**
added semantic release to a project, include:
- changelog
- release to GitHub
- automatic update version in core schemas
- automatic update version in package.json

Closes #30

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -7,11 +7,15 @@ git:
   submodules: false
 before_install:
   - git submodule update --init
-install:
   - npm install
+install:
+  - npm install @semantic-release/exec
+  - npm install @semantic-release/git
+  - npm install @semantic-release/changelog
+  - npm install travis-deploy-once
```
Comment (xferra) [inline .travis.yml]: I think they're also dev deps for the whole project, right?

File: package.json [hunk 2]
```diff
@@ -10,7 +10,8 @@
     "lint": "gulp lint",
     "import": "gulp import",
     "release": "gulp release && gulp test-release",
-    "update": "git fetch && git rebase && git submodule update --remote --merge"
+    "update": "git fetch && git rebase && git submodule update --remote --merge",
```
Comment (xferra) [inline package.json]: As we agreed, let's use 1 way of interacting with git, please rewrite it.

**Review by xferra — CHANGES_REQUESTED** [review]
FYI

File: .travis.yml (hunk 1 above)
Comment (romanrostislavovich) [reply .travis.yml]: done

File: package.json (hunk 2 above)
Comment (romanrostislavovich) [reply package.json]: done

**Comment by xferra** [comment]
Reimplemented in #39

**Pull request closed without merging**